### PR TITLE
[4.0] [A11y] lists scope and grid

### PR DESCRIPTION
**Pull request by brianteeman**
Continuing the work done (and merged) in #21087  See the description there for the reasons etc

### Summary of Changes
scope=row to title cells
scope=row to column headers

When a cell has a scope it is a th
When a cell is an action eg a checkbox to selectall then it is a td even if it is in a header row

Any styling changes are beyond the scope of this PR

### CODE REVIEW

File: administrator/components/com_categories/tmpl/categories/default.php [hunk 1]
```diff
@@ -73,65 +73,65 @@
...
-								<th style="width:1%" class="nowrap text-center d-none d-md-table-cell">
+								<th scope="col" style="width:1%" class="nowrap text-center d-none d-md-table-cell">
 									<?php echo HTMLHelper::_('searchtools.sort', '', 'a.lft', $listDirn, $listOrder, null, 'asc', 'JGRID_HEADING_ORDERING', 'icon-menu-2'); ?>
 								</th>
-								<th style="width:1%" class="text-center">
+								<td style="width:1%" class="text-center">
 									<?php echo HTMLHelper::_('grid.checkall'); ?>
-								</th>
+								</td>
 								<th style="width:1%" class="nowrap text-center">
```
Comment (Quy) [inline administrator/components/com_categories/tmpl/categories/default.php]: Add scope

File: administrator/components/com_finder/tmpl/filters/default.php [hunk 2]
```diff
@@ -33,25 +33,25 @@
 				<table class="table">
 					<thead>
 						<tr>
-							<th style="width:1%" class="nowrap text-center">
+							<td style="width:1%" class="nowrap text-center">
 								<?php echo JHtml::_('grid.checkall'); ?>
-							</th>
+							</td>
 							<th style="width:1%" class="nowrap">
```
Comment (Quy) [inline administrator/components/com_finder/tmpl/filters/default.php]: Add scope

File: administrator/components/com_finder/tmpl/maps/default.php [hunk 3]
```diff
@@ -34,25 +34,25 @@
 				<table class="table">
 					<thead>
 						<tr>
-							<th style="width:1%" class="text-center nowrap">
+							<td style="width:1%" class="text-center nowrap">
 								<?php echo JHtml::_('grid.checkall'); ?>
-							</th>
+							</td>
 							<th style="width:1%" class="text-center nowrap">
```
Comment (Quy) [inline administrator/components/com_finder/tmpl/maps/default.php]: Add scope

File: administrator/components/com_modules/tmpl/modules/default.php [hunk 4]
```diff
@@ -40,42 +40,42 @@
...
-						<th style="width:1%" class="nowrap text-center d-none d-md-table-cell">
+						<th scope="col" style="width:1%" class="nowrap text-center d-none d-md-table-cell">
 							<?php echo HTMLHelper::_('searchtools.sort', '', 'a.ordering', $listDirn, $listOrder, null, 'asc', 'JGRID_HEADING_ORDERING', 'icon-menu-2'); ?>
 						</th>
-						<th style="width:1%" class="nowrap text-center">
+						<td style="width:1%" class="nowrap text-center">
 							<?php echo HTMLHelper::_('grid.checkall'); ?>
-						</th>
+						</td>
 						<th style="width:1%; min-width:85px" class="nowrap text-center">
```
Comment (Quy) [inline administrator/components/com_modules/tmpl/modules/default.php]: Add scope

File: administrator/components/com_fields/tmpl/fields/default.php [hunk 5]
```diff
@@ -57,33 +57,33 @@
...
-								<th class="text-center">
+								<th scope="col" class="text-center">
 									<?php echo HTMLHelper::_('searchtools.sort', 'COM_FIELDS_FIELD_GROUP_LABEL', 'g.title', $listDirn, $listOrder); ?>
 								</th>
-								<th style="width:10%" class="nowrap d-none d-md-table-cell text-center">
+								<th scope="col" style="width:10%" class="nowrap d-none d-md-table-cell text-center">
 									<?php echo HTMLHelper::_('searchtools.sort', 'JGRID_HEADING_ACCESS', 'a.access', $listDirn, $listOrder); ?>
 								</th>
 								<?php if (Multilanguage::isEnabled()) : ?>
 									<th style="width:10%" class="nowrap d-none d-md-table-cell text-center">
```
Comment (Quy) [inline administrator/components/com_fields/tmpl/fields/default.php]: Add scope

File: administrator/components/com_fields/tmpl/fields/modal.php [hunk 6]
```diff
@@ -40,25 +40,25 @@
...
 						</th>
-						<th style="width:10%" class="nowrap d-none d-md-table-cell">
+						<th scope="col" style="width:10%" class="nowrap d-none d-md-table-cell">
 							<?php echo HTMLHelper::_('searchtools.sort', 'COM_FIELDS_FIELD_TYPE_LABEL', 'a.type', $listDirn, $listOrder); ?>
 						</th>
-						<th style="width:10%" class="nowrap d-none d-md-table-cell">
+						<th scope="col" style="width:10%" class="nowrap d-none d-md-table-cell">
 							<?php echo HTMLHelper::_('searchtools.sort', 'JGRID_HEADING_ACCESS', 'a.access', $listDirn, $listOrder); ?>
 						</th>
 						<th style="width:10%" class="nowrap d-none d-md-table-cell">
```
Comment (Quy) [inline administrator/components/com_fields/tmpl/fields/modal.php]: Add scope

File: administrator/components/com_finder/tmpl/maps/default.php [hunk 7]
```diff
@@ -34,25 +34,25 @@
...
-							<th style="width:1%" class="nowrap text-center">
-								<?php echo JText::_('COM_FINDER_HEADING_CHILDREN'); ?>
-							</th>
+								<th scope="col" style="width:1%" class="nowrap text-center">
+									<?php echo JText::_('COM_FINDER_HEADING_CHILDREN'); ?>
+								</th>
 							<?php endif; ?>
-							<th style="width:1%" class="nowrap text-center">
+							<th scope="col" style="width:1%" class="nowrap text-center">
                                 <span class="icon-publish" aria-hidden="true"></span>
```
Comment (Quy) [inline administrator/components/com_finder/tmpl/maps/default.php]: Indent

File: administrator/components/com_finder/tmpl/maps/default.php [hunk 8]
```diff
@@ -34,25 +34,25 @@
...
 								<span class="d-none d-md-inline"><?php echo JText::_('COM_FINDER_MAPS_COUNT_PUBLISHED_ITEMS'); ?></span>
 							</th>
-							<th style="width:1%" class="nowrap text-center">
+							<th scope="col" style="width:1%" class="nowrap text-center">
                                 <span class="icon-unpublish" aria-hidden="true"></span>
```
Comment (Quy) [inline administrator/components/com_finder/tmpl/maps/default.php]: Indent

File: administrator/components/com_users/tmpl/groups/default.php [hunk 9]
```diff
@@ -44,21 +44,21 @@
...
 									<?php echo HTMLHelper::_('grid.checkall'); ?>
-								</th>
-								<th class="nowrap">
+								</td>
+								<th scope="col" class="nowrap">
 									<?php echo HTMLHelper::_('searchtools.sort', 'COM_USERS_HEADING_GROUP_TITLE', 'a.title', $listDirn, $listOrder); ?>
 								</th>
-								<th style="width:10%" class="nowrap text-center">
+								<th scope="col" style="width:10%" class="nowrap text-center">
                                     <span class="icon-publish hasTooltip" aria-hidden="true" title="<?php echo Text::_('COM_USERS_COUNT_ENABLED_USERS'); ?>"></span>
```
Comment (Quy) [inline administrator/components/com_users/tmpl/groups/default.php]: Indent

File: administrator/components/com_users/tmpl/groups/default.php [hunk 10]
```diff
@@ -44,21 +44,21 @@
...
 									<span class="d-none d-md-inline"><?php echo Text::_('COM_USERS_COUNT_ENABLED_USERS'); ?></span>
 								</th>
-								<th style="width:10%" class="nowrap text-center">
+								<th scope="col" style="width:10%" class="nowrap text-center">
                                     <span class="icon-unpublish hasTooltip" aria-hidden="true" title="<?php echo Text::_('COM_USERS_COUNT_DISABLED_USERS'); ?>"></span>
```
Comment (Quy) [inline administrator/components/com_users/tmpl/groups/default.php]: Indent

File: administrator/components/com_installer/tmpl/languages/default.php [hunk 11]
```diff
@@ -66,9 +66,9 @@
 									<?php $onclick = 'document.getElementById(\'install_url\').value = \'' . $language->detailsurl . '\'; Joomla.submitbutton(\'install.install\');'; ?>
 									<input type="button" class="btn btn-secondary btn-sm" value="<?php echo JText::_('COM_INSTALLER_' . $buttonText . '_BUTTON'); ?>" onclick="<?php echo $onclick; ?>">
 								</td>
-								<td>
+								<th scope="row">
                                     <?php echo $language->name; ?>
```
Comment (Quy) [inline administrator/components/com_installer/tmpl/languages/default.php]: Indent

File: administrator/components/com_menus/tmpl/menus/default.php [hunk 12]
```diff
@@ -97,7 +97,7 @@
 								<td class="text-center">
 									<?php echo HTMLHelper::_('grid.id', $i, $item->id); ?>
 								</td>
-								<td>
+								<th scope="row">
```
Comment (Quy) [inline administrator/components/com_menus/tmpl/menus/default.php:100]: Change closing tag

File: administrator/components/com_contact/tmpl/contacts/modal.php [hunk 13]
```diff
@@ -118,15 +118,15 @@
 						<td class="text-center">
 							<span class="<?php echo $iconStates[$this->escape($item->published)]; ?>" aria-hidden="true"></span>
 						</td>
-						<td>
+						<th scope="row">
 							<a class="select-link" href="javascript:void(0)" data-function="<?php echo $this->escape($onclick); ?>" data-id="<?php echo $item->id; ?>" data-title="<?php echo $this->escape(addslashes($item->name)); ?>" data-uri="<?php echo $this->escape(ContactHelperRoute::getContactRoute($item->id, $item->catid, $item->language)); ?>" data-language="<?php echo $this->escape($lang); ?>">
 								<?php echo $this->escape($item->name); ?>
 							</a>
 							<?php echo $this->escape($item->name); ?></a>
 							<div class="small">
```
Comment (Quy) [inline administrator/components/com_contact/tmpl/contacts/modal.php]: Delete. Same as line 123.

File: administrator/components/com_banners/tmpl/clients/default.php [hunk 14]
```diff
@@ -113,7 +113,7 @@
 											<?php echo HTMLHelper::_('jgrid.published', $item->state, $i, 'clients.', $canChange); ?>
 										</div>
 									</td>
-									<td class="nowrap has-context">
+									<th class="nowrap has-context">
```
Comment (Quy) [inline administrator/components/com_banners/tmpl/clients/default.php]: Add scope

File: administrator/components/com_menus/tmpl/menus/default.php (hunk 12 above)
Comment (brianteeman) [reply administrator/components/com_menus/tmpl/menus/default.php:100]: done

**Comment by brianteeman** [comment]
Thanks @quy all done

File: administrator/components/com_menus/tmpl/items/default.php [hunk 15]
```diff
@@ -67,39 +67,39 @@
 									<?php echo HTMLHelper::_('searchtools.sort', '', 'a.lft', $listDirn, $listOrder, null, 'asc', 'JGRID_HEADING_ORDERING', 'icon-menu-2'); ?>
```
Comment (Quy) [inline administrator/components/com_menus/tmpl/items/default.php:67]: Add scope on line 66.

File: administrator/components/com_modules/tmpl/modules/default.php [hunk 16]
```diff
@@ -120,7 +120,7 @@
 								<?php echo HTMLHelper::_('grid.id', $i, $item->id); ?>
 							<?php endif; ?>
 						</td>
-						<td class="text-center">
+						<th class="text-center">
```
Comment (Quy) [inline administrator/components/com_modules/tmpl/modules/default.php]: Add scope and closing tag.

Comment (brianteeman) [reply administrator/components/com_modules/tmpl/modules/default.php]: almost. it shouldnt have been a th at all - its the next cell that is the th - fixed it though- THANKS

**Comment by Quy** [comment]
Under `System > Information > System Information`, apply `scope` there?

**Comment by brianteeman** [comment]
@quy yes but I will do that in a separate PR

**Comment by Quy** [comment]
Do the same for `Versions > Content History`.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on aec13ac327774cb7b56142b4bdab3997561ed8bf<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21157">issues.joomla.org/tracker/joomla-cms/21157</a>.</sub>

**Comment by Quy** [comment]
Do the same under `Permissions` tab.

**Comment by zwiastunsw** [comment]
I have tested this item :white_check_mark: successfully on aec13ac327774cb7b56142b4bdab3997561ed8bf<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21157">issues.joomla.org/tracker/joomla-cms/21157</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Comment by wojsmol** [comment]
@franz-wohlkoenig Please r5emove RTC label - PR has merge conflicts.;

**Comment by ghost** [comment]
Back on pending. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21157">issues.joomla.org/tracker/joomla-cms/21157</a>.</sub>

**Comment by ghost** [comment]
I set RTC having conflicting Files that its not lost and Maintainer decide if merge or not.

**Comment by brianteeman** [comment]
conflicts resolved

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21157">issues.joomla.org/tracker/joomla-cms/21157</a>.</sub>

**Comment by wilsonge** [comment]
thanks!

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**